File: codes/conexao.lua
function coroutineMain()
    --estabelecendo conexao via ZMQ
    context=simZMQ.ctx_new()
    -- responder eh o server side
    responder=simZMQ.socket(context,simZMQ.REP)
    local rc=simZMQ.bind(responder,'tcp://*:5555')
    if rc~=0 then error('failed bind') end

    while true do
        local rc,data=simZMQ.recv(responder,0)
        if data == "get_active_joint_angle" then
            local joint_angles = angulo_junta(active_joint_handles)
            simZMQ.send(responder,joint_angles,0)
        end
    end
end

File: codes/conexao.py
import zmq
import numpy as np
import spatialmath as sm
import roboticstoolbox as rtb

context = zmq.Context()

#  Socket to talk to server
print("Connecting to hello world server…")
socket = context.socket(zmq.REQ)
socket.connect("tcp://localhost:5555")

#  Do 10 requests, waiting each time for a response

print("Sending request ...")
# socket.send(b"Hello")
socket.send(b"get_active_joint_angle")

#  Get the reply.
# message = socket.recv()
joint_angles_str = socket.recv()